[BUG] autoreact cog doesn't handle quickly deleted messages

**Describe the bug**

If a message is sent in a channel with autoreacts configured (e.g. #labpics in Homelab) and then very quickly deleted (e.g. in the case of a non-compliant #labpics post, quickly removed by the enforcer cog), the autoreact cog throws an exception when it can't find the message to react to.

**To Reproduce**
Steps to reproduce the behavior:
1. Send message in autoreact-configured channel
2. Delete it very quickly
3. See exception

**Expected behavior**

Error is caught and handled.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem -->

**Additional context**

```py
[2023-05-16 09:26:31] [ERROR] discord.client: Ignoring exception in on_message
Traceback (most recent call last):
  File "/home/redbot/.local/lib/python3.11/site-packages/discord/client.py", line 441, in _run_event
    await coro(*args, **kwargs)
  File "/redbot/data/cogs/CogManager/cogs/autoreact/autoreact.py", line 43, in on_message
    await message.add_reaction(reaction)
  File "/home/redbot/.local/lib/python3.11/site-packages/discord/message.py", line 1098, in add_reaction
    await self._state.http.add_reaction(self.channel.id, self.id, emoji)
  File "/home/redbot/.local/lib/python3.11/site-packages/discord/http.py", line 740, in request
    raise NotFound(response, data)
discord.errors.NotFound: 404 Not Found (error code: 10008): Unknown Message
```

--- a/autoreact/autoreact.py
+++ b/autoreact/autoreact.py
@@ -1,5 +1,6 @@
 """discord red-bot autoreact"""
 import asyncio
+import logging
 from typing import Generator, Optional
 
 import discord
@@ -10,6 +11,7 @@ from redbot.core.utils.predicates import ReactionPredicate
 
 CUSTOM_CONTROLS = {"⬅️": prev_page, "➡️": next_page}
 
+log = logging.getLogger("red.rhomelab.autoreact")
 
 class AutoReactCog(commands.Cog):
     """AutoReact Cog"""
@@ -40,7 +42,16 @@ class AutoReactCog(commands.Cog):
         if str(message.channel.id) in channels.keys():  # process special reactions
             channel_reactions = channels[str(message.channel.id)]
             for reaction in channel_reactions:
-                await message.add_reaction(reaction)
+                try:
+                    await message.add_reaction(reaction)
+                except discord.errors.NotFound:
+                    log.info(
+                        "Could not react to message %s in channel %s (%s) as the message was not found." +
+                        "Maybe the message was deleted?",
+                        message.id,
+                        message.channel.name,
+                        message.channel.id
+                    )
 
         # Do not continue if channel is whitelisted
         if message.channel.id in whitelisted_channels:
@@ -49,7 +60,16 @@ class AutoReactCog(commands.Cog):
         for phrase in reactions.keys():
             if phrase in message.content.lower().split():
                 for emoji in reactions[phrase]:
-                    await message.add_reaction(emoji)
+                    try:
+                        await message.add_reaction(emoji)
+                    except discord.errors.NotFound:
+                        log.info(
+                            "Could not react to message %s in channel %s (%s) as the message was not found." +
+                            "Maybe the message was deleted?",
+                            message.id,
+                            message.channel.name,
+                            message.channel.id
+                        )
 
     # Command groups
 

